src/main/java/cl/transbank/webpay/example/controller/WebpayController.java

```diff
@@ -0,0 +1,55 @@
+package cl.transbank.webpay.example.controller;
+
+import cl.transbank.webpay.exception.TransactionCreateException;
+import cl.transbank.webpay.webpayplus.WebpayPlus;
+import cl.transbank.webpay.webpayplus.model.CreateWebpayPlusTransactionResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+@Controller
+public class WebpayController {
+    private static Logger log = LoggerFactory.getLogger(WebpayController.class);
+
+    @RequestMapping(value = "/", method = RequestMethod.GET)
+    public ModelAndView index() {
+        return new ModelAndView("index");
+    }
+
+    @RequestMapping(value = "/webpayplus", method = RequestMethod.GET)
+    public ModelAndView webpayplus() {
+        String buyOrder = String.valueOf(new Random().nextInt(Integer.MAX_VALUE));
+        String sessionId = String.valueOf(new Random().nextInt(Integer.MAX_VALUE));
+        double amount = 1000;
+        String returnUrl = "http://localhost:8081/webpayplus-end";
```

Comment:
This won't work if I'm using a browser in a different computer. Which mean we can't deploy this on heroku for example. 

Isn't there a way to figure out the "base URL" of a spring boot controller? If you have an `HttpServletRequest` object, you can figure that out using `getScheme()`, `getServerName(), `getServerPort(), `getContextPath()`, etc. 

Or if that's not the way to go at least give the option to override the base url with a environment variable. The point is, let people run the demo on servers or access the demo from machines other than localhost.

Reply:
Fixed